python/cudf/cudf/core/multiindex.py

```diff
@@ -47,7 +47,8 @@ def __init__(
 
         # early termination enables lazy evaluation of codes
         if "source_data" in kwargs:
-            source_data = kwargs["source_data"].reset_index(drop=True)
+            kwargs["source_data"].reset_index(drop=True, inplace=True)
```

Comment:
You can't modify the input here, need to make a shallow copy and then modify that inplace.